ibm/service/vpc/resource_ibm_is_share.go

```diff
@@ -181,6 +181,13 @@ func ResourceIbmIsShare() *schema.Resource {
 										Computed:    true,
 										Description: "If `true`:- The VPC infrastructure performs any needed NAT operations.- `floating_ips` must not have more than one floating IP.If `false`:- Packets are passed unchanged to/from the network interface,  allowing the workload to perform any needed NAT operations.- `allow_ip_spoofing` must be `false`.- If the virtual network interface is attached:  - The target `resource_type` must be `bare_metal_server_network_attachment`.  - The target `interface_type` must not be `hipersocket`.",
 									},
+									"protocol_state_filtering_mode": {
```

Comment:
missing test cases and potentially data source changes, and test file for data source

Reply:
only the VNI and instance template API response contains protocol_state_filtering_mode. So only the data sources for VNI and VNI collection and template has the changes. 
Test cases are added